test/1.4/lib/T_io_memory.dml

```diff
@@ -29,6 +42,30 @@ bank a {
         return 0xaa;
     }
 }
+subdevice ab {
+    implement io_memory {
+        is function_io_memory;
+    }
+    bank b {
+        is function_mapped_bank;
+        param function = 0xb;
+        method read(uint64 address, uint64 enabled_bytes, void *user)
+            -> (uint64) throws {
+            return 0xab;
+        }
+    }
+    port function_io_memory_port {
+        implement io_memory {
+            is function_io_memory;
+        }
+    }
+    bank function_io_memory_bank {
+        param use_io_memory = false;  // not using bank_io_memory
+        implement io_memory {
+            is function_io_memory;
+        }
+    }
```

Comment:
To make this test completely exhaustive, this subdevice should feature a subdevice of its own with function-mapped banks, and then the test should confirm that those are not being considered.

Reply:
Alternatively, instead of a sub-subdevice, add a test case confirming that the main device is not considering the banks of its subdevices. That tests the same logic.

Reply:
Also, consider wrapping one of the banks or ports in a group.